### PR TITLE
CLDR-8666 Implementation for Vote Report Resolver

### PR DESCRIPTION
**Pull request by srl295**
CLDR-8666 Vote Report Resolver and tests

- also make VoterReportStatus abstract instead of interface
- move ReportStatusUpdater into org.unicode.cldr.util (not web specific)
- tests added, but they depend on CLDR-15765 to be safe (so disabled for now)

- [ ] This PR completes the ticket.

### CODE REVIEW

**Comment by srl295** [comment]
see if this makes some sense

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/ReportsDB.java [hunk 1]
```diff
@@ -5,9 +5,6 @@
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
-import java.time.ZoneOffset;
-import java.time.ZonedDateTime;
-import java.time.format.DateTimeFormatter;
```
Comment (srl295) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/ReportsDB.java:10]: unrelated cleanup of imports

**Review by srl295 — COMMENTED** [review]
Note that the test does pass if enabled, but would conflict with other tests.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VoterReportStatus.java [hunk 2]
```diff
@@ -1,19 +1,39 @@
 package org.unicode.cldr.util;
 
 import java.util.EnumSet;
+import java.util.Set;
 
 /**
  * This interface is for objects which can expose information on which reports have been completed.
+ * @param T the type for each voter. Must be an Integer.
```
Comment (btangmu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/VoterReportStatus.java:8]: "Must be an Integer" -- then why generic?

Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VoterReportStatus.java:8]: See the note. It was originally generic because it didn't need to be more specific. But voteresolver's user Id is fixed as an int. I was trying to minimize churn by not removing the generic just yet.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VoterReportStatus.java [hunk 3]
```diff
@@ -35,5 +55,43 @@ public ReportStatus mark(ReportId r, boolean asComplete, boolean asAcceptable) {
...
+     * Update a Resolver for a particular Report. The resolver will be cleared.
+     * Note that T must be an Integer for this to succeed.
+     * @param l locale
+     * @param r which report
+     * @param userList set of users
+     * @param status the source
+     * @param res which
+     * @return resolver
+     */
+    public VoteResolver<ReportAcceptability>
```
Comment (btangmu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/VoterReportStatus.java:79]: ordinarily we have "VoteResolver<String>" where the strings are candidate items...

is the plan to use "VoteResolver<ReportAcceptability>" to resolve votes between "acceptable" and "notAcceptable"?

Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VoterReportStatus.java:79]: yes, that's exactly what it does.  it's a voteresolver applied to an enum.

Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VoterReportStatus.java:79]: it means that all the usual voting rules apply- per-organization, strengths, etc.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/VoterReportStatus.java (hunk 2 above)
Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/VoterReportStatus.java:8]: @btangmu  i added a note to https://unicode-org.atlassian.net/browse/CLDR-15765 to also remove the generic when that ticket goes in (post close).

**Comment by srl295** [comment]
Translating the test case:

- fr: Nobody Votes.
    - Status: missing
- fr: Meta vetter votes ‘acceptable’
    - Status: approved
    - Value: acceptable
- fr: Guest guest votes ‘not acceptable’
    - Status:approved
    - Value: acceptable
- fr: Wikimedia vetter votes ‘not acceptable’
    - Status: approved
    - Value: not acceptable (5 vs 4)
- fr: Guest guest abstains
    - Status: provisional
    - Value: acceptable (arbitrary?)


- ssy: Guest votes notAcceptable
    - Status: unconfirmed
    - Value: not Acceptable


With this change, the dashboard, etc, will be able to calculate and display the 'final' result for each of the reports:
- Missing (nobody has voted)
- Acceptable (with status: approved, provisional, etc…)
- NotAcceptable (with status: approved, provisional, etc…)

**Review by macchiati — APPROVED** [review]
